nixos/modules/services/network-filesystems/glusterfs.nix

```diff
@@ -5,6 +5,22 @@ with lib;
 let
   inherit (pkgs) glusterfs rsync;
 
+  tlsCmd = if cfg.enableTLS then
+  ''
+    mkdir -p /var/lib/glusterd
+    touch /var/lib/glusterd/secure-access
+  ''
+  else
+  ''
+    rm -f /var/lib/glusterd/secure-access
+  '';
+
+  restartTriggers = [
+    config.environment.etc."ssl/glusterfs.pem".source
+    config.environment.etc."ssl/glusterfs.key".source
```

Comment:
> What is a better approach?

Giving it a path (that's not in the nix store).

For example, `nixops` has a `keys` functionality that places secret keys into a special directory that doesn't survive reboots.